File: 7.reverse-integer.py
# @leet start
class Solution:
    def reverse(self, x: int) -> int:
        sign: int = [1, -1][x < 0]
        x: int = abs(x)
        rev: int = 0 # Placeholder for reversed number
        while x:
            x, mod = divmod(x, 10)
            rev = rev * 10 + mod
            if rev > 2**31 - 1: 
                return 0 
        return sign * rev


      # @leet end

File: 7.reverse-integer.cpp
// @leet start

#include <cmath>
#include <iostream>
class Solution {
public:
    int reverse(int x) {
        long reversed = 0;
        int pop;
        while (x != 0){
            pop = x % 10;
            x /= 10;
            reversed = (reversed*10)+pop;
            if (reversed < (pow(2,31)*-1) || reversed > pow(2,31))
            {
                return 0;
            }
        }
        return reversed;
    }
};
// @leet end
